.github/workflows/update-gitops.yaml

```diff
@@ -0,0 +1,111 @@
+name: Update GitOps repository
+
+on:
+  workflow_call:
+    inputs:
+      packages:
+        description: 'Packages to update'
+        required: true
+        type: string
+      environment:
+        description: 'Environment e.g. beta, prod, main used to define path to kustomize overlay'
+        required: true
+        type: string
+      image-prefix:
+        description: 'Prefix of image e.g. thinc-org/cugetreg'
+        required: true
+        type: string
+      gitops-repository:
+        description: 'Target GitOps repository'
+        required: true
+        type: string
+      gitops-ref:
+        description: 'Target GitOps ref'
+        type: string
+        default: refs/heads/master
+      container-registry:
+        description: 'Container registry e.g. ghcr.io'
+        type: string
+        default: ghcr.io
+      mode:
+        description: 'Mode of the action, pr or commit'
+        type: string
+        default: pr
+    secrets:
+      GH_TOKEN:
+        description: 'GitHub token used to checkout GitOps repository and open PR'
+        required: true
+
+jobs:
+  update-gitops:
+    name: Update GitOps and Open PR
+
+    runs-on: ubuntu-latest
+
+    steps:
+      - name: Checkout Repo
+        uses: actions/checkout@v3
+        with:
+          repository: ${{ inputs.gitops-repository }}
+          ref: ${{ inputs.gitops-ref }}
+          token: ${{ secrets.GH_TOKEN }}
+
+      - name: Setup Kustomize
+        uses: multani/action-setup-kustomize@v1
+        with:
+          version: 5.0.0
+
+      - name: Update kustomize configuration
+        run: |
+          for row in $(echo "$PACKAGES" | jq -r '.[] | @base64'); do
+            _jq() {
+              echo ${row} | base64 --decode | jq -r ${1}
+            }
+            NAME=$(_jq '.name')
+            IMAGE_TAG=$(_jq '.imageTag')
+            PREFIX=${{ inputs.container-registry }}/${{ inputs.image-prefix }}
+            bash -c "cd $NAME/overlays/${{ inputs.environment }} && kustomize edit set image $NAME=$PREFIX/$NAME:$IMAGE_TAG"
+            echo "${NAME}:${IMAGE_TAG} is updated"
+          done
+        env:
+          PACKAGES: ${{ inputs.packages }}
+
+      - name: Show Git Status
+        run: git status
+
+      - name: Prepare Pull Request body
+        id: pr-body
+        run: |
+          echo "DATE=$(date +'%d/%m/%Y')" >> $GITHUB_OUTPUT
+          UPDATED_PACKAGES=$(echo "$PACKAGES" | jq -r '.[] | "- \(.name): \(.imageTag)"')
+          echo "UPDATED_PACKAGES<<EOF" >> $GITHUB_ENV
+          echo "$UPDATED_PACKAGES" >> $GITHUB_ENV
+          echo "EOF" >> $GITHUB_ENV
```

Comment:
why do we use this as ENV instead of OUTPUT like DATE na?

Reply:
Because of this issue. https://github.com/octokit/request-action/issues/6 https://github.com/github/docs/issues/21529

Workaround https://trstringer.com/github-actions-multiline-strings/

TLDR; GitHut output fuck with multiple line output